src/lib/Metadata/Finalizers/BaseMembersFinalizer.cpp

```diff
@@ -30,53 +30,57 @@ inheritBaseMembers(
     RecordInterface const& base,
     AccessKind const A)
 {
+    auto&& members = allMembers(derived);
+
     if (A == AccessKind::Public)
     {
         // When a class uses public member access specifier to derive from a
         // base, all public members of the base class are accessible as public
         // members of the derived class and all protected members of the base
         // class are accessible as protected members of the derived class.
         // Private members of the base are never accessible unless friended.
-        inheritBaseMembers(derivedId, derived.Public, base.Public);
-        inheritBaseMembers(derivedId, derived.Protected, base.Protected);
+        inheritBaseMembers(derivedId, derived.Public, base.Public, members);
```

Comment:
Is this safe? `members` is a const-view of the derived members, but the function is changing the set of members. It's OK to ask for all members there again, but this is supposed to be a lazy operation (unless there's some bug somewhere else).